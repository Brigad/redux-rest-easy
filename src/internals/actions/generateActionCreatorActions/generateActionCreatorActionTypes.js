import generateActionType from '../../utils/generateActionType';

const generateActionCreatorActionTypes = (resourceName, actionName) => ({
  REQUEST: generateActionType(resourceName, actionName, 'REQUEST'),
  RECEIVE: generateActionType(resourceName, actionName, 'RECEIVE'),
  FAIL: generateActionType(resourceName, actionName, 'FAIL'),
  RECEIVE_FROM_CACHE: generateActionType(
    resourceName,
    actionName,
    'RECEIVE_FROM_CACHE',
  ),
});

export default generateActionCreatorActionTypes;
