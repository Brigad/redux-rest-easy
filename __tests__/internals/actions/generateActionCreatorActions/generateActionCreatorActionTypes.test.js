import generateActionCreatorActionTypes from '../../../../src/internals/actions/generateActionCreatorActions/generateActionCreatorActionTypes';

describe('generateActionCreatorActionTypes', () => {
  test('only path', () => {
    expect(generateActionCreatorActionTypes('fruits', 'eat')).toMatchSnapshot();
  });
});
