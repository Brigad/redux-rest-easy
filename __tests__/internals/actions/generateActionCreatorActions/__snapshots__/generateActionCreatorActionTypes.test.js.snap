// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateActionCreatorActionTypes only path 1`] = `
Object {
  "FAIL": "@@rest-easy/fruits/eat/FAIL",
  "RECEIVE": "@@rest-easy/fruits/eat/RECEIVE",
  "RECEIVE_FROM_CACHE": "@@rest-easy/fruits/eat/RECEIVE_FROM_CACHE",
  "REQUEST": "@@rest-easy/fruits/eat/REQUEST",
}
`;
